src/App.tsx

```diff
@@ -1,26 +1,276 @@
+
+  const hasComplitedTodos = () => {
+    if (todoList.length === 0) {
+      return true;
+    }
+
+    return todoList.some(todo => todo.completed);
+  };
+
+  const handleSetInputElement = (newInputElement: HTMLInputElement | null) => {
+    currentInputElementRef.current = newInputElement;
+  };
+
+  const callFocus = () => {
+    const element = currentInputElementRef.current;
+
+    if (element) {
+      setTimeout(() => {
+        element.focus();
+      }, 0);
+    }
+  };
+
+  const processSuccessfulTodos = async (
+    todos: Todo[],
+    serviceCall: (t: Todo) => Promise<unknown>,
+    message: ErrorMessages,
+  ) => {
+    const results = await Promise.allSettled(
+      todos.map(todoToRemove => {
+        return serviceCall(todoToRemove);
+      }),
+    );
+
+    const failedTodos = results
+      .map((result, i) => (result.status === 'rejected' ? todos[i] : null))
+      .filter(Boolean);
+
+    if (failedTodos.length) {
+      setErrorMessage(message);
+    }
+
+    return todos.filter(todo => !failedTodos.includes(todo));
+  };
+
+  const onAddTodo = async (title: string) => {
+    const newTodo: Todo = {
+      id: 0,
+      title: title,
+      completed: false,
+      userId: USER_ID,
+    };
+
+    setTempTodo({ ...newTodo });
+
+    try {
+      const requestResult = await todosService.add(newTodo);
+
+      setTempTodo(null);
+      setTodoList(current => [...current, requestResult]);
+
+      return requestResult;
+    } catch {
+      setErrorMessage(ErrorMessages.addError);
+      setTempTodo(null);
+
+      return null;
+    } finally {
+      callFocus();
+    }
+  };
+
+  const onRemoveTodo = async (todoToRemove: Todo) => {
+    setTodosToLoading(prev => [...prev, todoToRemove]);
+
+    try {
+      await todosService.remove(todoToRemove);
+      setTodoList(prev => prev.filter(todo => todo.id !== todoToRemove.id));
+    } catch {
+      setErrorMessage(ErrorMessages.deleteError);
+    } finally {
+      callFocus();
+    }
+  };
+
+  const removeAllComplited = async () => {
+    const todosToRemove = todoList.filter(todo => todo.completed);
+
+    setTodosToLoading(todosToRemove);
+    const successTodosRequest = await processSuccessfulTodos(
+      todosToRemove,
+      todosService.remove,
+      ErrorMessages.deleteError,
+    );
+
+    setTodosToLoading([]);
+    setTodoList(prev =>
+      prev.filter(todo => !successTodosRequest.includes(todo)),
+    );
+    callFocus();
+  };
+
+  const onUpdateTodo = async (newTodo: Todo) => {
+    setTodosToLoading(prev => [...prev, newTodo]);
+    try {
+      const updatedTodo = await todosService.update(newTodo);
+
+      setTodoList(prev =>
+        prev.map(todo => (todo.id === updatedTodo.id ? updatedTodo : todo)),
+      );
+    } catch {
+      setErrorMessage(ErrorMessages.updateError);
+      throw new Error();
+    } finally {
+      setTodosToLoading([]);
+    }
+  };
+
+  const isToggleAll = () => {
+    return todoList.every(todo => todo.completed);
+  };
```

Comment:
Mind the naming

For instance: 
```suggestion
  const areAllTodosCompleted = () => {
    return todoList.every(todo => todo.completed);
  };
```

etc.